rsmtool/reporter.py

```diff
@@ -599,84 +599,72 @@ def create_report(self,
         if test_file_location is None:
-            test_file_location = configuration.get('pred_file_location')
+            test_file_location = config.get('pred_file_location')
 
         # raise error if location is still None
         if test_file_location is None:
             raise KeyError('Could not find `test_file_location` or `pred_file_location` '
                            'in Configuration object. Please make sure you have included '
                            'one of these parameters in the configuration object.')
 
-        # set the environment variables we want
-        os.environ['EXPERIMENT_ID'] = experiment_id
-        os.environ['DESCRIPTION'] = description
-        os.environ['CONTEXT'] = context
-        os.environ['TRAIN_FILE_LOCATION'] = train_file_location
-        os.environ['TEST_FILE_LOCATION'] = test_file_location
-        os.environ['OUTPUT_DIR'] = csvdir
-        os.environ['FIGURE_DIR'] = figdir
-        os.environ['MODEL_NAME'] = model_name
-        os.environ['MODEL_TYPE'] = model_type
-        os.environ['STANDARDIZE_FEATURES'] = '1' if standardize_features else '0'
-        os.environ['SCALED'] = '1' if use_scaled_predictions else '0'
-        os.environ['EXCLUDE_ZEROS'] = '1' if exclude_zero_scores else '0'
-        os.environ['LENGTH_COLUMN'] = '' if length_column is None else length_column
-        os.environ['H2_COLUMN'] = ('' if second_human_score_column is None
-                                   else second_human_score_column)
-        os.environ['MIN_ITEMS'] = ('0' if min_items_per_candidate is None
-                                   else str(min_items_per_candidate))
-        os.environ['FEATURE_SUBSET_FILE'] = ('' if feature_subset_file is None
-                                             else feature_subset_file)
-        os.environ['JAVASCRIPT_PATH'] = javascript_path
-
-        # we define separate groups to allow future flexibility in defining
-        # what groups are used for descriptives and evaluations
-        os.environ['GROUPS_FOR_DESCRIPTIVES'] = '%%'.join(subgroups)
-        os.environ['GROUPS_FOR_EVALUATIONS'] = '%%'.join(subgroups)
-
-        # pass in other useful information to the report
-        os.environ['FILE_FORMAT'] = file_format
-        os.environ['USE_THUMBNAILS'] = '1' if use_thumbnails else '0'
-        os.environ['SKLL_OBJECTIVE'] = '' if skll_objective is None else skll_objective
-        os.environ['PREDICT_EXPECTED_SCORES'] = '1' if predict_expected_scores else '0'
+        # if the features subset file is None, just use empty string
+        feature_subset_file = ('' if config.get('feature_subset_file') is None
+                               else config['feature_subset_file'])
+
+        # if the min items is None, just set it to zero
+        min_items = (0 if config['min_items_per_candidate'] is None
+                     else config['min_items_per_candidate'])
+
+        environ_config = {'EXPERIMENT_ID': config['experiment_id'],
```

Comment:
We are doing this funny thing where we first capitalize all field names in .json and then convert them to lower case again as we read in the values in header. It's ok to keep as is.